language-guides/spring-data-neo4j/spring-data-neo4j.adoc

```diff
@@ -34,129 +34,206 @@ When developing with Neo4j, please use JDK 8 and your favorite IDE.
+public interface MovieRepository extends Neo4jRepository<Movie, Long> {
+
+    @Query("MATCH (m:Movie {title: {title}}) RETURN m LIMIT 1")
+	Movie findByTitle(@Param("title") String title);
 
-  @Query("MATCH (m:Movie)<-[rating:RATED]-(user)
-          WHERE id(m) = {movieId} RETURN rating")
-  List<Rating> getRatings(@Param("movieID") Long movieId);
+	Collection<Movie> findByTitleLike(@Param("title") String title);
 
-  List<Movie> findByTitle(String title);
+    @Query("MATCH (m:Movie)<-[r:ACTED_IN]-(a:Person) RETURN m,r,a LIMIT {limit}")
+	Collection<Movie> graph(@Param("limit") int limit);
 }
 ----
 
+In the code below, the MovieService class injects the MovieRepository in using a constructor.
+The next blocks of code are the implementations for the methods defined in the MovieRepository class.
+Annotating these methods with @Transactional and read-only will allow enterprise applications using causal clustering to improve performance by routing these transactions to replica servers and focus the write traffic to the core servers.
+Each method calls the defined method in MovieRepository and returns results formatted in the expected return type.
+
 .Use repository
 [source,java]
 ----
-@Autowired MovieRepository repo;
+@Service
+public class MovieService {
+
+	private final MovieRepository movieRepository;
+	public MovieService(MovieRepository movieRepository) {
+		this.movieRepository = movieRepository;
+	}
+
+    @Transactional(readOnly = true)
+    public Movie findByTitle(String title) {
+        Movie result = movieRepository.findByTitle(title);
+        return result;
+    }
 
-List<Movie> movie = repo.findByTitle("The Matrix");
+    @Transactional(readOnly = true)
+    public Collection<Movie> findByTitleLike(String title) {
+        Collection<Movie> result = movieRepository.findByTitleLike(title);
+        return result;
+    }
 
-List<Rating> ratings = repo.getRatings(movieId);
+	@Transactional(readOnly = true)
+	public Map<String, Object>  graph(int limit) {
+		Collection<Movie> result = movieRepository.graph(limit);
+		return toD3Format(result);
+	}
 ----
 
-We provide a https://github.com/neo4j-examples?q=sdn[number of example projects] using Spring Data Neo4j, Spring Boot and Web-Frameworks like Angular.js.
+For a more thorough walkthrough of the code, see the resources linked below.
+We also provide a number of https://github.com/neo4j-examples?q=sdn[example projects] using Spring Data Neo4j, Spring Boot, and web frameworks like Angular.js on GitHub.
 
 :maven-sdn: http://search.maven.org/#search|ga|1|a%3A%22spring-data-neo4j%22
 
 === Resources
 
 [cols="1,4"]
 |===
-| icon:code-fork[] Project | http://projects.spring.io/spring-data-neo4j
-| icon:user[] Authors | The Neo4j, http://graphaware.com/neo4j-experts/[GraphAware] and Pivotal teams.
+| icon:code-fork[] Projects | https://projects.spring.io/spring-data-neo4j/[Spring Data Neo4j], https://projects.spring.io/spring-boot/[Spring Boot]
```

Comment:
not sure if we need boot in here?

Reply:
I also think we should remove this because it is neither our project nor the project in focus for this page.